Lib/fontTools/misc/bezierTools.py

```diff
@@ -753,10 +765,389 @@ def calcCubicPoints(a, b, c, d):
+    return not (xDiff <= 0.0 and yDiff <= 0.0)
+
+
+def lineLineIntersections(s1, e1, s2, e2):
+    """Finds intersections between two line segments.
+
+    Args:
+        s1, e1: Coordinates of the first line as 2D tuples.
+        s2, e2: Coordinates of the second line as 2D tuples.
+
+    Returns:
+        A list of ``Intersection`` objects, each object having ``pt``, ``t1``
+        and ``t2`` attributes containing the intersection point, time on first
+        segment and time on second segment respectively.
+
+    Examples::
+
+        >>> a = lineLineIntersections( (310,389), (453, 222), (289, 251), (447, 367))
+        >>> len(a)
+        1
+        >>> intersection = a[0]
+        >>> intersection.pt
+        (374.44882952482897, 313.73458370177315)
+        >>> (intersection.t1, intersection.t2)
+        (0.45069111555824454, 0.5408153767394238)
+    """
+    ax, ay = s1
+    bx, by = e1
+    cx, cy = s2
+    dx, dy = e2
+    if math.isclose(cx, dx) and math.isclose(ax, bx):
+        return []
+    if math.isclose(cy, dy) and math.isclose(ay, by):
+        return []
+    if math.isclose(cx, dx) and math.isclose(cy, dy):
+        return []
+    if math.isclose(ax, bx) and math.isclose(ay, by):
+        return []
+    if math.isclose(bx, ax):
+        x = ax
+        slope34 = (dy - xy) / (dx - cx)
+        y = slope34 * (x - cx) + cy
+        pt = (x, y)
+        return [
+            Intersection(
+                pt=pt, t1=_line_t_of_pt(s1, e1, pt), t2=_line_t_of_pt(s2, e2, pt)
+            )
+        ]
+    if math.isclose(cx, dx):
+        x = cx
+        slope12 = (by - ay) / (bx - ax)
+        y = slope12 * (x - ax) + ay
+        pt = (x, y)
+        return [
+            Intersection(
+                pt=pt, t1=_line_t_of_pt(s1, e1, pt), t2=_line_t_of_pt(s2, e2, pt)
+            )
+        ]
+
+    slope12 = (by - ay) / (bx - ax)
+    slope34 = (dy - cy) / (dx - cx)
+    if math.isclose(slope12, slope34):
+        return []
+    x = (slope12 * ax - ay - slope34 * cx + cy) / (slope12 - slope34)
+    y = slope12 * (x - ax) + ay
+    pt = (x, y)
+    if _both_points_are_on_same_side_of_origin(
+        pt, e1, s1
+    ) and _both_points_are_on_same_side_of_origin(pt, s2, e2):
+        return [
+            Intersection(
+                pt=pt, t1=_line_t_of_pt(s1, e1, pt), t2=_line_t_of_pt(s2, e2, pt)
+            )
+        ]
+    return []
+
+
+def _alignment_transformation(segment):
+    start = segment[0]
+    end = segment[-1]
+    m = Offset(-start[0], -start[1])
+    endpt = m.transformPoint(end)
+    angle = math.atan2(endpt[1], endpt[0])
+    return m.reverseTransform(Identity.rotate(-angle))
+
+
+def _curve_line_intersections_t(curve, line):
+    aligned_curve = _alignment_transformation(line).transformPoints(curve)
+    if len(curve) == 3:
+        a, b, c = calcCubicParameters(*aligned_curve)
+        intersections = solveQuadratic(a[0], b[0], c[0])
+        intersections.extend(solveQuadratic(a[1], b[1], c[1]))
```

Comment:
Hm, are you sure it's correct to solve for _both_ x=0 and y=0 like you seem to be doing here? Should you not only solve for y=0, since with your transformation you have rotated the curve and the intersecting line to coincide with the horizontal x axis, thus you want to know the t values for which y=0?

See the similar function inside bezierTools called `splitCubic`: this does basically the same thing as yours, minus the previous rotation. If you do your transform, then call splitCubic with `where` parameter set to 0 and `isHorizontal` set to True, meaning `where` is interpreted as a Y coordinate, then you see it's only calling `solveCubic` once (using the y cubic parameters), not twice.